src/lib/auxiliary_database/filtered_external_transition.ml

```diff
@@ -104,27 +104,39 @@ let of_transition ~tracked_participants {With_hash.data= external_transition; _}
         { Transactions.user_commands= []
         ; fee_transfers= []
         ; coinbase= Currency.Amount.zero } ~f:(fun acc_transactions -> function
-      | User_command checked_user_command ->
+      | User_command checked_user_command -> (
           let user_command = User_command.forget_check checked_user_command in
-          if
+          let should_include_transaction user_command participants =
             List.exists
               (User_command.accounts_accessed user_command)
-              ~f:(Public_key.Compressed.Set.mem tracked_participants)
-          then
-            { acc_transactions with
-              user_commands= user_command :: acc_transactions.user_commands }
-          else acc_transactions
+              ~f:(Public_key.Compressed.Set.mem participants)
+          in
+          match tracked_participants_opt with
+          | Some tracked_participants
+            when not
+                   (should_include_transaction user_command
+                      tracked_participants) ->
+              acc_transactions
+          | _ ->
```

Comment:
I had to think about this code for a while. Instead of having `tracked_participants_opt` be an option, can you make it something like ```[`All | `Some of Public_key.Compressed.Set.t]```?